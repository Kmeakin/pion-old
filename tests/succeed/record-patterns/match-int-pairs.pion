//~ run = "${PION} elab ${FILE}"

{
let f1 : Int -> Int -> Int
= fun x y => match (x, y) {
    (0, a) => a,
    (b, 0) => b,
    (1, c) => c,
    (d, 1) => d,
    (_, _) => 5,
};

let f2 : Int -> Int -> Int
= fun x y => {
    let p = (x, y);
    match p {
        (0, a) => a,
        (b, 0) => b,
        (1, c) => c,
        (d, 1) => d,
        (_, _) => 5,
    }
};

{}
}
