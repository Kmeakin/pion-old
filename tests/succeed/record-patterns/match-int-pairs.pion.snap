exit status: 0
stdout:
let f1 : Int -> Int -> Int =
    fun (x : Int) (y : Int) => let a : (Int, Int) = (x, y);
    match a._0 {
        0 => let a : Int = a._1;
        a,
        1 => match a._1 {
            0 => let b : Int = a._0;
            b,
            1 => let c : Int = a._1;
            c,
            _ => let c : Int = a._1;
            c,
        },
        _ => match a._1 {
            0 => let b : Int = a._0;
            b,
            1 => let d : Int = a._0;
            d,
            _ => 5,
        },
    };
let f2 : Int -> Int -> Int =
    fun (x : Int) (y : Int) => let p : (Int, Int) = (x, y);
    match p._0 {
        0 => let a : Int = p._1;
        a,
        1 => match p._1 {
            0 => let b : Int = p._0;
            b,
            1 => let c : Int = p._1;
            c,
            _ => let c : Int = p._1;
            c,
        },
        _ => match p._1 {
            0 => let b : Int = p._0;
            b,
            1 => let d : Int = p._0;
            d,
            _ => 5,
        },
    };
() : ()

stderr:
